Pypi prep

mbolivar opened the pull request:
A couple of patches I would like to merge before tagging v0.1.0 and uploading to PyPI.

carlescufi commented on README.rst:
```diff
@@ -1,81 +1,99 @@
+repository, you can run the ``west`` package as a module::
 
-To run the test suite, use:
+  PYTHONPATH=/path/to/west/repository/src python3 -m west
```
I guess this only works on *nix. On Windows one would have to resort to things like [these](https://superuser.com/a/1049433). In particular the easiest is probaby `cmd /C "set PYTHONPATH=/path/to/west/repo && python3 -m west"`

carlescufi commented on README.rst:
```diff
@@ -1,81 +1,98 @@
+To use another manifest repository (optionally with ``--mr
+some-manifest-branch``)::
+
+  west init -u https://example.com/your-manifest-repository.git
```
This made me realize that a common usecase would be to specify both a non-default west location and also a non-default manifest repo. Maybe we could add a combined option like:
`west init -uw https://github.com/myuser`, and west would automatically append `/west.git` and `/manifest.git` to the URL in order to fetch those.

mbolivar replied:
Eew :).

Sure, we could do something like that. Maybe `-b` or `--url-base`? I'm assuming this is post-0.1.0.

carlescufi replied:
>Eew :)
You mean eew `-uw`? `-b` is fine with me :)

mbolivar replied:
Oh, just seems a bit weird to me personally, but I see the value for short command lines.

carlescufi replied:
I think it makes sense, but this can wait if you don't want to add it now.

mbolivar commented on README.rst:
```diff
@@ -1,81 +1,98 @@
+a directory of your choosing::
 
-West is distributed in two pieces:
+  mkdir zephyr-dir && cd zephyr-dir
```
We could, for sure, but I didn't want this to be ambiguous with the actual Zephyr tree that gets cloned by `west fetch`.

Do you want me to make this change?

carlescufi replied:
Right, in that case how about `mkdir zephyrproject && cd zephyrproject`?

carlescufi commented on README.rst:
```diff
@@ -1,81 +1,98 @@
+------------
 
-DO NOT INSTALL WITH "python3 setup.py install".
+Install West's bootstrapper with pip::
```
I wonder if we should capitalize West, or leave it lowercase `west`.

mbolivar replied:
Sure, I'll lowercase it.

carlescufi commented on README.rst:
```diff
@@ -1,81 +1,98 @@
-  Git repositories, including its own.
-- West is in its experimental stages and is moving quickly, meaning
-  users need to stay on HEAD.
+- ``west fetch`` clones the repositories in the manifest, creating
```
maybe comment here that `west fetch` is run by the freshly cloned `west` code

mbolivar replied:
I'm not sure what this means ... ``west fetch`` isn't run by the code, not sure how code can run something. Sorry to be dense, could you please clarify?

carlescufi replied:
Just after installing `west` from PyPi with `pip`, you can run `west fetch`. The Python script that actually executes all the required `git fetch` operations is the one in the `west/` folder, not the one installed by pip, isn't it? This is no big deal, just a small clarification that might or might not be applicable.

carlescufi replied:
Nice, thank you

carlescufi commented on README.rst:
```diff
@@ -1,81 +1,98 @@
-development mode, clone this repository and run this from the top
-level directory:
+West has multiple sub-commands. After running ``west init``, you can
+run them from anywhere under ``zephyr-dir``.
```
`zephyrproject`

carlescufi commented on README.rst:
```diff
@@ -1,81 +1,98 @@
+Install West's bootstrapper with pip::
 
-Please use pip to install in development mode as documented below.
+  pip3 install west
```
One of the things I feel are extremely useful and sorely miss on Windows cmd line is git bash autocomplete. I wonder if we could reuse their script somehow and actually install it during bootstrapping.

mbolivar replied:
Taking a serious look at autocomplete on all the platforms is on my wishlist as well -- should we start backlogging this in the issues for this repo?

carlescufi replied:
Yes actually. I will create one right now.

ulfalizer commented on README.rst:
```diff
@@ -1,81 +1,98 @@
+To install everything in development mode instead, clone the West
+repository and run this from the top level directory::
+
+  pip3 install -e .
```
Would be nice to have a note on how you run `west` after running this command (`python -m west`). A plain `west` doesn't work.

mbolivar replied:
It should... what goes wrong?

ulfalizer replied:
Never mind. It's working.

Maybe a note could be added about needing `.local/bin/` in PATH.

mbolivar commented:
Most changes addressed, some questions still remain though, @ulfalizer @carlescufi

mbolivar commented:
> Approving as-is, although there are improvements that can be postponed. For now the biggest priority is to get this up on PyPi.

I've added a --base-url option to the bootstrap code and addressed README comments.

Should be good to go -- @ulfalizer?

mbolivar commented:
> Should be good to go

I've also added a Python 2 version check and reordered the commits so the one resetting to 0.1.0 can be tagged properly.

carlescufi commented:
Is this ready for merging now?

mbolivar commented:
> do a point release when that's done

Sorry, I am wrong, no need to do another release for self-update, as that's non-bootstrap and won't get sent to pypi.

Moving west and the manifest would require changes to what's in pypi if we do it after making the first 0.1.0 upload.

carlescufi commented:
@ulfalizer please re-review so we can merge this and then rebase your PR on top

carlescufi commented:
>The change to having west and manifest under a single top level directory would be pretty significant as well, since that would break existing installations and affect the bootstrapper. We should decide on that before asking for wider use.

That one is part of the bootstrap but in a separate PR, I suggest that we merge this first then rebase that PR on top.

As per publishing to pypi, I would publish as soon as this one is merged just because I'm nervous someone may take the `west` name otherwise.

mbolivar commented:
> That one is part of the bootstrap but in a separate PR, I suggest that we merge this first then rebase that PR on top.
> 
> As per publishing to pypi, I would publish as soon as this one is merged just because I'm nervous someone may take the west name otherwise

+1 to both.

ulfalizer commented on README.rst:
```diff
@@ -1,81 +1,110 @@
+  pip3 install -e .
+
+(On Linux, make sure ``~/.local/bin`` is in your ``PATH`` if you go
+this route.)
```
Maybe this could be added at the end, within the parenthesis: "Alternatively, you can run ``west`` with ``python -m west``."
